src/ex4/dist/app.js

```diff
@@ -0,0 +1,102 @@
+const addButton = document.getElementById("list-item-submit")
+const textInput = document.getElementById("list-item-input")
+const allLists = document.getElementById("list")
+const counterLists = document.getElementById("counter")
+const errorMessage = document.getElementById("error")
+const regex = /^[0-9]*$/
+
+const itemC = new ItemClient()
+
+class Main {
+  constructor() {
+    this.name = "Main"
+  }
+  init() {
+    addButton.addEventListener("click", function () {
+      validation()
+    })
+  }
+}
+function addNewList(name) {
+  const li = document.createElement("li")
+  const hr = document.createElement("hr")
+  const deleteButton = document.createElement("button")
+  deleteButton.setAttribute("class", "dltbtn")
+  deleteButton.setAttribute("id", textInput.value)
+  deleteButton.innerHTML = "X"
+
+  deleteButton.addEventListener("click", async function (e) {
+    li.remove()
+    hr.remove()
+    await itemC.deleteItem(e.target.id)
+    countLists()
+  })
+  li.addEventListener("mouseover", function () {
+    deleteButton.style.display = "block"
+    deleteButton.style.transition = "all 0.5s ease-in-out"
+  })
+  li.addEventListener("mouseout", function () {
+    deleteButton.style.display = "none"
+    deleteButton.style.transition = "all 0.5s ease-in-out"
+  })
+  li.appendChild(document.createTextNode(name))
+  li.appendChild(deleteButton)
+  allLists.appendChild(li)
+  allLists.appendChild(hr)
+  textInput.value = ""
+}
+
+async function validation() {
```

Comment:
1. This validation function does more than just validating.- it also performs the server call and handles the response
2. Adding a todo is now broken, in case the input is a string it tries to fetch from the API
![image](https://user-images.githubusercontent.com/36815567/174824679-733f89c2-dc3a-4f19-ab2b-213e2c2407e0.png)